lib/filterx/func-sdata.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2024 Szilard Parrag
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "func-sdata.h"
+#include "filterx/object-primitive.h"
+#include "filterx/filterx-eval.h"
+#include "filterx/object-string.h"
+
+#include <string.h>
+
+
+FilterXObject *
+filterx_simple_function_is_sdata_from_enterprise(FilterXExpr *s, GPtrArray *args)
+{
+  if (args == NULL || args->len != 1)
+    {
+      filterx_simple_function_argument_error(s, "Requires exactly one argument", FALSE);
+      return NULL;
+    }
+
+  FilterXObject *obj = filterx_typecast_string(NULL, args);
+  if (!obj)
+    return NULL;
+  const gchar *str_value = filterx_string_get_value(obj, NULL);
+  if (!str_value)
+    return NULL;
```

Comment:
I think the filterx_object_extract_string() function would make this simpler, #229

Reply:
Good point, switched to them!